tests/integration/test_sns.py

```diff
@@ -2113,3 +2115,115 @@ def test_publish_to_gcm(self, sns_client):
+                    "Effect": "Allow",
+                    "Principal": {"Service": "firehose.amazonaws.com"},
+                    "Action": "sts:AssumeRole",
+                },
+                {
+                    "Effect": "Allow",
+                    "Principal": {"Service": "sns.amazonaws.com"},
+                    "Action": "sts:AssumeRole",
+                },
+            ],
+        }
+
+        role = create_role(RoleName=role_name, AssumeRolePolicyDocument=json.dumps(trust_policy))
+
+        iam_client.attach_role_policy(
+            RoleName=role_name,
+            PolicyArn="arn:aws:iam::aws:policy/AmazonKinesisFirehoseFullAccess",
+        )
+
+        iam_client.attach_role_policy(
+            RoleName=role_name, PolicyArn="arn:aws:iam::aws:policy/AmazonS3FullAccess"
+        )
+        subscriptionRoleArn = role["Role"]["Arn"]
+
+        if is_aws_cloud():
+            time.sleep(10)
+
+        s3_create_bucket(Bucket=bucket_name)
+
+        stream = firehose_client.create_delivery_stream(
+            DeliveryStreamName=stream_name,
+            DeliveryStreamType="DirectPut",
+            S3DestinationConfiguration={
+                "RoleARN": subscriptionRoleArn,
+                "BucketARN": f"arn:aws:s3:::{bucket_name}",
+                "BufferingHints": {"SizeInMBs": 1, "IntervalInSeconds": 60},
+            },
+        )
+
+        wait_for_delivery_stream_ready(stream_name)
+
+        topic = sns_create_topic(Name=topic_name)
+        sns_subscription(
+            TopicArn=topic["TopicArn"],
+            Protocol="firehose",
+            Endpoint=stream["DeliveryStreamARN"],
+            Attributes={"SubscriptionRoleArn": subscriptionRoleArn},
+            ReturnSubscriptionArn=True,
+        )
+
+        message = json.dumps({"message": "hello world"})
+        messageAttributes = {
+            "testAttribute": {"DataType": "String", "StringValue": "valueOfAttribute"}
+        }
+        sns_client.publish(
+            TopicArn=topic["TopicArn"], Message=message, MessageAttributes=messageAttributes
+        )
+
+        def validate_content():
+            files = s3_client.list_objects(Bucket=bucket_name)["Contents"]
+            f = BytesIO()
+            s3_client.download_fileobj(bucket_name, files[0]["Key"], f)
+            content = to_str(f.getvalue())
+
+            sns_message = json.loads(content.split("\n")[0])
+
+            assert "Type" in sns_message
+            assert "MessageId" in sns_message
+            assert "Message" in sns_message
+            assert "Timestamp" in sns_message
+
+            assert message == sns_message["Message"]
+
+        retries = 5
+        sleep = 1
+        sleep_before = 0
+        if is_aws_cloud():
+            retries = 25
+            sleep = 10
+            sleep_before = 10
+
+        retry(validate_content, retries=retries, sleep_before=sleep_before, sleep=sleep)
+        firehose_client.delete_delivery_stream(DeliveryStreamName=stream_name)
```

Comment:
nit: should we manage the delivery stream in a fixture as well, to assure the cleanup of the resource? We could use the `wait_for_delivery_stream` in the fixture as well.